make readme a bit more clear (#28)
MrSimonEmms/gitpod-k3s-guide: README.md

Before starting the installation process, you need:

- Target building resources.
  - Ubuntu 20.04/22.04 machine(s) with SSH credentials. 
    - At least one, but also the script can also work for multiple nodes. The hostname of each node can be called `node0`, `node1`, etc.

>     - `node0` must have ports 22 (SSH), 80 (HTTP), 443 (HTTPS) and 6443 (Kubernetes) exposed. `node0` must have a static public IP address that allows domain name to point to. `node<x>` must have post 22 (SSH) exposed. All nodes are better to be in the same vlan so they can communicate with each other.

Reviewer: I'm not sure this is entirely correct. In my setup, there is a load balancer on all nodes HTTP and HTTPS. Also, if you use the `HA_CLUSTER`, then you can connect to the Kubernetes API on subsequent nodes.

I think for simplicity, just saying "expose these four ports" is ok.

Author: Yes, I agree it should be more general than to just talking about my own setting. I have applied the changes and also updated this part a bit.

>     - At least 4 cores, 16GB RAM and 50GB storage.

Reviewer: ```suggestion
    - At least 4 cores, 16GB RAM and 100GB storage.
```

We had some people on Discord complaining below 100GB storage

>   - A domain and some wildcard subdomains managed by cloudflare (free), gcp, or route53 [see price](https://aws.amazon.com/route53/pricing/). please see [here](https://www.gitpod.io/docs/configure/self-hosted/latest/installing-gitpod) about "DNS and TLS configured". These DNS services will have and manage free letsencrypt certificates for you. If you choose not using these commercial DNS services, you will need to use self-signed certificates and manage it by yourself.

Reviewer: ```suggestion
  - A domain and some wildcard subdomains managed by Cloudflare (free), GCP, or Route53 [see price](https://aws.amazon.com/route53/pricing/). Please see the "DNS and TLS configured" section in the [Gitpod docs](https://www.gitpod.io/docs/configure/self-hosted/latest/installing-gitpod) for more information. These DNS services will have and manage free Let's Encrypt certificates for you. If you choose not to use these commercial DNS services, you will need to use self-signed certificates and manage them manually.
```

- A `.env` file or environment variables with basic details about the environment.
  - We provide an example of such file [here](.env.example)

> - Building environment:

Reviewer: ```suggestion
- Building environment. You can either:
```

>   - build on a local linux machine - needs to install kubectl, Helm, K3sup, also check and clean up `~/.kube/` if there was config file before. 

Reviewer: ```suggestion
  - Build on a local Linux machine - needs to install kubectl, Helm, K3sup. You may need to clean your `${HOME}/.kube` directory if there was a previous `gitpod-k3s` entry. 
```

>   - Or use [Docker](https://docs.docker.com/engine/install) installed on your machine and Docker file is at .gitpod/gitpod.Dockerfile

Reviewer: ```suggestion
  - Use [Docker](https://docs.docker.com/engine/install) installed on your machine and Docker file is at .gitpod/gitpod.Dockerfile
```

>   - Even better, use [Gitpod workspace](https://gitpod.io/#https://github.com/MrSimonEmms/gitpod-k3s-guide)😀.

Reviewer: ```suggestion
  - Even better, use a [Gitpod workspace](https://gitpod.io/#https://github.com/MrSimonEmms/gitpod-k3s-guide)😀.
```

>   - Or even better, use [Gitpod workspace](https://gitpod.io/#https://github.com/MrSimonEmms/gitpod-k3s-guide) 😀

Reviewer: ```suggestion
  - Even better, use [Gitpod workspace](https://gitpod.io/#https://github.com/MrSimonEmms/gitpod-k3s-guide) 😀
```
